corehq/apps/custom_data_fields/edit_model.py

```diff
@@ -291,6 +292,11 @@ def save_profiles(self):
             )
             if not created and obj.has_users_assigned:
                 refresh_es_for_profile_users.delay(self.domain, obj.id)
+                user_ids_to_sync = obj.user_ids_assigned()
+                from corehq.apps.users.models import CommCareUser
+                for user_id in user_ids_to_sync:
+                    user = CommCareUser.get_by_user_id(user_id)
+                    sync_usercases_if_applicable(user, spawn_task=False)
```

Comment:
Looking at this method it only requires the User ID so we can change it take the User ID instead of the user object. This would save us having to load the user objects from the DB: https://github.com/dimagi/commcare-hq/blob/0f6535972646bb06ac694d966a1f45101d570c44/corehq/apps/callcenter/tasks.py#L52-L55

Also, I think this should spawn the task rather than doing the update in the request.

Reply:
I load the user objects because this if-statement uses the user object to check if the project has the feature flags enabled. I wasn't sure how to get around this. Any suggestions?

https://github.com/dimagi/commcare-hq/blob/0f6535972646bb06ac694d966a1f45101d570c44/corehq/apps/callcenter/tasks.py#L51-L52

Reply:
good point, you could pass in the domain name and use `Domain.get_by_name(domain)` to get the domain object. Also considering this is potentially doing multiple users you could take a slightly different approach:

```python
def bulk_sync_usercases_if_applicable(domain, user_ids):
    domain_obj = Domain.get_by_name(domain)
    if domain_obj.call_center_config.enabled or domain_obj.usercase_enabled:
        for user_id in user_ids:
            sync_usercases_task.delay(user_id)

bulk_sync_usercases_if_applicable(obj.definition.domain, obj.user_ids_assigned())
```